pluto.jl

```diff
@@ -13,37 +13,37 @@ end
+如今许多科学领域都用到了编程，这些领域内的科学家们通常不得不为自己的项目编写定制的代码。然而，对于大多数科学家来说，计算机科学不是他们的专业领域；他们不可避免地需要学习编程。我认为自己就是其中之一。虽然我们可能对编程的**软件**方面比较熟悉，但是我们对计算机**硬件**如何影响代码性能甚至很少有基本的认识。
 
-The aim of this tutorial is to give non-professional programmers a *brief* overview of the features of modern hardware that you must understand in order to write fast code. It will be a distillation of what have learned the last few years. This tutorial will use Julia because it allows these relatively low-level considerations to be demonstrated easily in a high-level, interactive language.
+本教程的目标是向业余编程选手提供现代硬件特性的**简明**概述，其中只包含了编写快速代码所必须了解的特性。这将是对过去几年所学知识的升华。本教程将使用 Julia， 因为它能够使用高级交互式语言轻松地演示这些相对低层级的考虑。
 
-## What this notebook is not
-#### This is not a guide to the Julia programming language
-To write fast code, you must first understand your programming language and its idiosyncrasies. But this is *not* a guide to the Julia programming language. I recommend reading the [performance tips section](https://docs.julialang.org/en/v1/manual/performance-tips/) of the Julia documentation.
+## 这份手册不是什么
+#### 这不是关于 Julia 编程语言的指南 
+为了编写快速代码，你必须首先了解你的编程语言及其特性。但这**不是**关于 Julia 编程语言的指南。我建议阅读 Julia 文档中的[性能建议](https://docs.juliacn.com/latest/manual/performance-tips/) 节。
 
-#### This is not an explanation of specific data structures or algorithms
-Besides knowing your language, you must also know your own code to make it fast. You must understand the idea behind big-O notation, why some algorithms are faster than others, and how different data structures work internally. Without knowing *what an `Array` is*, how could you possibly optimize code making use of arrays?
+#### 这不是关于特定数据结构或算法的说明
+为了加速代码，除了要弄明白编程语言，你也必须理解你自己的代码。你必须了解大 O 记号，为什么一些算法比其他算法快, 以及不同数据结构内部如何组织。若是不知道**什么是`Array`**，又怎么能够使用数组优化代码？
 
-This too, is outside the scope of this paper. However, I would say that as a minimum, a programmer should have an understanding of:
+这同样超出了本文的范围。然而，我想说的是，一个编程选手至少应该了解：
 
-* How a binary integer is represented in memory
-* How a floating point number is represented in memory (learning this is also necessary to understand computational inaccuracies from floating point operations, which is a must when doing scientific programming)
-* The memory layout of a `String` including ASCII and UTF-8 encoding
-* The basics of how an `Array` is structured, and what the difference between a dense array of e.g. integers and an array of references to objects are
-* The principles behind how a `Dict` (i.e. hash table) and a `Set` works
+* 二进制整数在内存中的表示方式
+* 浮点数在内存中的表示方式（学习这点也是必要的，因为这有助于理解浮点运算的计算误差，而浮点运算是科学计算中不可或缺的部分）
+* `String` 的内存布局，包括 ASCII 和 UTF-8 编码
+* 关于`Array` 结构化的基本知识，以及由数构成的稠密数组与对象引用数组之间的区别
+* `Dict`（例如，哈希表）和 `Set` 背后的工作原理
```

Comment:
"i.e." 表示 "即" 的意思

```suggestion
* 哈希表（如字典 `Dict` 和集合 `Set`）背后的工作原理
```

是的... `Set` 也是基于 hash 表实现的...